test/compression_cases_test.go

```diff
@@ -0,0 +1,108 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/internal/stubserver"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+	"google.golang.org/grpc/status"
+)
+
+type mockCompressor struct {
+	grpc.Compressor
+}
+
+func newMockCompressor() grpc.Compressor {
+	return &mockCompressor{grpc.NewGZIPCompressor()}
+}
+
+func (c *mockCompressor) Type() string {
+	return "mock"
+}
+
+type mockDecompressor struct {
+	grpc.Decompressor
+}
+
+func newMockDecompressor() grpc.Decompressor {
+	return &mockDecompressor{grpc.NewGZIPDecompressor()}
+}
+
+func (d *mockDecompressor) Type() string {
+	return "mock"
+}
+
+func TestCompressionCases(t *testing.T) {
+	cases := []struct {
+		desc           string
+		clientUseMock  bool
+		serverUseMock  bool
+		expectedStatus codes.Code
+	}{
+		{
+			desc:           "Client and Server use mock compression",
```

Comment:
`t.Run` needs the first parameter as the test name. The test name should be like a variable name in snake case. I would recommend changing this field name from `desc` to `name` and changing the values to small phrases in snake case, e.g. name `client_server_custom_compression`.

Reply:
I think we can use any field in the struct as the first parameter, either it might be a name or desc. 
I see that in this [test](https://github.com/grpc/grpc-go/blob/1b1230bb69fba5c6c1653602d576b17025fbf50f/trace_test.go#L44) we used desc

Reply:
We follow the Google Style Guide to ensure uniform style across the repository. The style guide mentions how subtests should be named: https://google.github.io/styleguide/go/decisions#subtest-names